bson/src/main/org/bson/codecs/pojo/PojoCodecImpl.java

```diff
@@ -48,41 +43,28 @@
     private final CodecRegistry registry;
     private final PropertyCodecRegistry propertyCodecRegistry;
     private final DiscriminatorLookup discriminatorLookup;
-    private final ConcurrentMap<ClassModel<?>, Codec<?>> codecCache;
     private final boolean specialized;
 
     PojoCodecImpl(final ClassModel<T> classModel, final CodecRegistry codecRegistry,
                   final List<PropertyCodecProvider> propertyCodecProviders, final DiscriminatorLookup discriminatorLookup) {
         this.classModel = classModel;
-        this.registry = fromRegistries(fromCodecs(this), codecRegistry);
+        this.registry = codecRegistry;
         this.discriminatorLookup = discriminatorLookup;
-        this.codecCache = new ConcurrentHashMap<ClassModel<?>, Codec<?>>();
```

Comment:
The codecCache wasn't useful and was essentially replaced with a cache in the PropertyCodecRegistryImpl